### DISCUSSION TITLE
bug in core

### DISCUSSION THREAD

**Discussion by DanSkareda** [SpenceKonde/megaTinyCore#527] [Library discussion]
I did a clean install of the Arduino IDE 1.8.16. Using the "json" link I installed the current megaTinyCore. In the sketch folder I copied the unzipped current version of OneWireNg. In its Examples folder is the DallasTemperature sample project. I opened that and did a compilation for attiny3216/Optiboot (it doesn't work in general for any circuit). The compilation ended with errors:

C:\Users\radek\AppData\Local\Temp\cccalydd.ltrans0.ltrans.o: In function UartClass::begin(unsigned long, unsigned int)': <artificial>:(.text+0x8ee): undefined reference to operator delete(void*, unsigned int)'
collect2.exe: error: ld returned 1 exit status
exit status 1

In the spring I also tried it for attiny3216, everything was fine then. The library author's opinion on this problem is here:

https://github.com/pstolarz/OneWireNg/issues/20#issuecomment-918427756

Here is all discussion:

https://github.com/pstolarz/OneWireNg/issues/20

**Comment by DanSkareda** [comment]
Today I modified the megaTinyCore according to the advice in the link and then compiled the project without errors and Dallas DS18b20 works. :-)

**Comment by SpenceKonde** [comment]
I have commented in that discussion; there has been at least one and possibly two prior issues relating to `new` which is a black box to me. Can someone *please* give me a definitive reference for what functions the new.* files should contain, and documentation of what they do? Or is this just some undocumented black box that gets passed around,. added to in secret, and then reported as bugs when other core authors don't divine that `new` has been updated?

**Comment by SpenceKonde** [comment]
#528 - please use that for further discussion, as this is clearly a defect in the core, and should be tracked as such.
